diff --git a/tests/test_nativetypes.py b/tests/test_nativetypes.py
index 8c8525251..136908180 100644
--- a/tests/test_nativetypes.py
+++ b/tests/test_nativetypes.py
@@ -160,3 +160,13 @@ def test_macro(env):
     result = t.render()
     assert result == 2
     assert isinstance(result, int)
+
+
+def test_block(env):
+    t = env.from_string(
+        "{% block b %}{% for i in range(1) %}{{ loop.index }}{% endfor %}"
+        "{% endblock %}{{ self.b() }}"
+    )
+    result = t.render()
+    assert result == 11
+    assert isinstance(result, int)
