diff --git a/httpx/_utils.py b/httpx/_utils.py
index 7fe827d..e6671a8 100644
--- a/httpx/_utils.py
+++ b/httpx/_utils.py
@@ -109,7 +109,6 @@ def peek_filelike_length(stream: typing.Any) -> int | None:
             # then put it back in place.
             offset = stream.tell()
             length = stream.seek(0, os.SEEK_END)
-            stream.seek(offset)
         except (AttributeError, OSError):
             # Not even that? Sorry, we're doomed...
             return None
